lua/telescope/builtin/__lsp.lua

```diff
@@ -97,51 +97,74 @@ lsp.outgoing_calls = function(opts)
   calls(opts, "to")
 end
 
----@type { [string]: fun(results: table, items: table, opts: table): table, table }
+--- convert `item` type back to something we can pass to `vim.lsp.util.jump_to_location`
+--- stopgap for pre-nvim 0.10 - after which we can simply use the `user_data`
+--- field on the items in `vim.lsp.util.locations_to_items`
+---@param item vim.lsp.util.locations_to_items.ret
+---@param offset_encoding string|nil utf-8|utf-16|utf-32
+---@return lsp.Location
+local function item_to_location(item, offset_encoding)
+  local line = item.lnum - 1
+  local character = vim.lsp.util._str_utfindex_enc(item.text, item.col, offset_encoding) - 1
+  return {
+    uri = vim.uri_from_fname(item.filename),
+    range = {
+      start = {
+        line = line,
+        character = character,
+      },
+      ["end"] = {
+        line = line,
+        character = character,
+      },
+    },
+  }
+end
+
+---@alias telescope.lsp.list_or_jump_action
+---| "textDocument/references"
+---| "textDocument/definition"
+---| "textDocument/typeDefinition"
+---| "textDocument/implementation"
+
+-- luacheck: push ignore
+---@type { [telescope.lsp.list_or_jump_action]: fun(items: vim.lsp.util.locations_to_items.ret[], opts: table): vim.lsp.util.locations_to_items.ret[] }
+-- luacheck: pop
 local action_handlers = {
-  ["textDocument/references"] = function(results, items, opts)
+  ["textDocument/references"] = function(items, opts)
```

Comment:
it doesnt make sense to have an "action_handers" table here, it has one entry, why would we want to construct a table for that.

we could just do this in `apply_action_handler`
```lua
if action == "textDocument/references" then
  -- stuff from function
end
return locations, items
```

Reply:
its a fine idea for an abstraction but we dont need it here, we have 1 entry, we can still introduce an abstraction if we have like 3, 4, 5 entries